examples/node/domstubs.js

```diff
@@ -91,24 +91,26 @@ DOMElement.prototype = {
   },
 
   toString: function DOMElement_toString() {
-    var attrList = [];
-    for (i in this.attributes) {
-      attrList.push(i + '="' + xmlEncode(this.attributes[i]) + '"');
+    var str = '<' + this.nodeName;
+    if (this.nodeName === 'svg:svg') {
+      str += ' xmlns:xlink="http://www.w3.org/1999/xlink"' +
+             ' xmlns:svg="http://www.w3.org/2000/svg"';
     }
+    for (var i in this.attributes) {
+      str += ' ' + i + '="' + xmlEncode(this.attributes[i]) + '"';
+    }
+
+    str += '>';
 
     if (this.nodeName === 'svg:tspan' || this.nodeName === 'svg:style') {
-      var encText = xmlEncode(this.textContent);
-      return '<' + this.nodeName + ' ' + attrList.join(' ') + '>' +
-             encText + '</' + this.nodeName + '>';
-    } else if (this.nodeName === 'svg:svg') {
-      var ns = 'xmlns:xlink="http://www.w3.org/1999/xlink" ' +
-               'xmlns:svg="http://www.w3.org/2000/svg"'
-      return '<' + this.nodeName + ' ' + ns + ' ' + attrList.join(' ') + '>' +
-             this.childNodes.join('') + '</' + this.nodeName + '>';
+      str += xmlEncode(this.textContent);
     } else {
-      return '<' + this.nodeName + ' ' + attrList.join(' ') + '>' +
```

Comment:
> It is conceptually as simple as replacing += extraData with .write(extraData)

Yeah, the extent of the current changes is good. I just want to replace `str +=` to `buf.push()` verbatim, which will fit the agenda above.

> One of the representations is a sequence of substrings...

I understand. But we can produce really large string and I don't want JS JITs to make the call how to handle this data internally.